src/os/src/detail/osTime.c

```diff
@@ -596,11 +602,11 @@ int32_t taosTimeCountInterval(int64_t skey, int64_t ekey, int64_t interval, char
   localtime_r(&t, &tm);
   int emon = tm.tm_year * 12 + tm.tm_mon;
 
-  if (unit == 'y') {
-    interval *= 12;
+  if (slidingUnit == 'y') {
+    sliding *= 12;
   }
 
-  return (int32_t)(emon/interval - smon/interval) + 1;
+  return (int32_t)(emon/sliding - smon/sliding) + 1;
```

Comment:
The effect of code is the same as old. Truncate by subtracting remainder within a sliding window, using a unified approach.